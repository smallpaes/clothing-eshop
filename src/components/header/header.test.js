import { shallow } from 'enzyme';

import Header from './header.component';
import CartDropdownContainer from '../cart-dropdown/cart-dropdown.container';

describe('Header component', () => {
  let wrapper;
  let mockSignOutStart;

  beforeEach(() => {
    mockSignOutStart = jest.fn();
    const mockPros = {
      currentUser: {
        uid: 123
      },
      hidden: true,
      signOutStart: mockSignOutStart
    };

    wrapper = shallow(<Header { ...mockPros } />);
  });

  it('Should render Header component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('currentUser is present', () => {
    it('Should render sign out button', () => {
      expect(
        wrapper
          .find('OptionLink')
          .at(2)
          .text()
      ).toBe('SIGN OUT');
    });

    it('Should signOutStart got triggered when sign out link is clicked', () => {
      wrapper
        .find('OptionLink')
        .at(2)
        .simulate('click');

      expect(mockSignOutStart.mock.calls.length).toBe(1);
    });
  });

  it('Should hide CartDropdownContainer component when hidden prop is true', () => {
    expect(wrapper.exists(CartDropdownContainer)).toBe(false);
  });

  describe('currentUser is null', () => {
    const mockNewPros = {
      currentUser: null,
      hidden: false,
      signOutStart: mockSignOutStart
    };

    const newWrapper = shallow(<Header { ...mockNewPros } />);

    it('Should render signin button', () => {
      expect(newWrapper.exists('[to="/signin"]')).toBe(true);
    });

    it('Should render CartDropdownContainer component when hidden prop is false', () => {
      expect(newWrapper.exists(CartDropdownContainer)).toBe(true);
    });
  });
});