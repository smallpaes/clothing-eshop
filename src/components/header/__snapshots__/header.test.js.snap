// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render Header component 1`] = `
<styled.div>
  <Styled(Link)
    to="/"
  >
    <ForwardRef(SvgCrown)
      className="logo"
    />
  </Styled(Link)>
  <OptionsContainer>
    <OptionLink
      to="/shop"
    >
      SHOP
    </OptionLink>
    <OptionLink
      to="/contact"
    >
      CONTACT
    </OptionLink>
    <OptionLink
      as="div"
      onClick={[MockFunction]}
    >
      SIGN OUT
    </OptionLink>
    <Connect(CartIcon) />
  </OptionsContainer>
</styled.div>
`;
